src/OtcNexus.sol

```diff
@@ -0,0 +1,603 @@
+                if (_paymentTokenAmount > rfs.amount1) revert OtcNexus__Amount1TooHigh();
+            } else {
+                address aggregator = priceFeeds[rfs.tokensAccepted[index]];
+                if (aggregator == address(0)) revert OtcNexus__InvalidAggregatorAddress();
+                (, int256 price, , , ) = AggregatorV3Interface(aggregator).latestRoundData();
+                _amount0 = OtcMath.getTakerAmount0(
+                    _paymentTokenAmount,
+                    rfs.usdPrice * 1e8,
+                    uint256(price)
+                );
+            }
+        } else if (expectedRfsType == RfsType.DYNAMIC) {
+            address aggregatorTaker = priceFeeds[rfs.tokensAccepted[index]];
+            if (aggregatorTaker == address(0)) revert OtcNexus__InvalidAggregatorAddress();
+            (, int256 priceTaker, , , ) = AggregatorV3Interface(aggregatorTaker).latestRoundData();
+
+            address aggregatorMaker = priceFeeds[rfs.token0];
+            if (aggregatorMaker == address(0)) revert OtcNexus__InvalidAggregatorAddress();
+            (, int256 priceMaker, , , ) = AggregatorV3Interface(aggregatorMaker).latestRoundData();
+            _amount0 = OtcMath.getTakerAmount0(
+                _paymentTokenAmount,
+                (uint256(priceMaker) * rfs.priceMultiplier) / 100,
+                uint256(priceTaker)
+            );
+        } else {
+            revert OtcNexus__InvalidRfsType();
+        }
+    }
+
+    function computeRewards(
+        address maker,
+        address taker,
+        uint256 _amount0Bought,
+        uint256 _amount0Total
+    ) internal {
+        uint256 percentageBought = (_amount0Bought * OtcToken(otcToken).decimals()) / _amount0Total; //Calculate the percentage of the total amount bought with the decimals of the token so less computations and we can just send this amount
+        unchecked {
+            userRewards[taker] = percentageBought;
+            userRewards[maker] = percentageBought * 3; //3x the amount of the buyer
+        }
+    }
+
+    /**
+     * @dev This function calculates the fee to be taken from a specified amount based on a
+     * given fee percentage. It also calculates the amount left after the fee is taken.
+     *
+     * The fee is calculated as a percentage of the input amount, where the percentage is
+     * expressed as a value out of 10,000 (i.e., basis points). For example, if the
+     * `applicableFeePercentage` is 50, then the fee will be 0.5% of the input amount.
+     *
+     * @param applicableFeePercentage the fee rate expressed as basis points (i.e., hundredths of a percent)
+     * @param _paymentTokenAmount the amount from which the fee is to be taken
+     * @return feeAmount the calculated fee amount
+     * @return amountAfterFee the amount that remains after the fee is subtracted
+     *
+     * Requirements:
+     *
+     * - `applicableFeePercentage` must be less than or equal to 10,000 (100%).
+     */
+    function takeFee(
```

Comment:
the name of this function should be calculateFee as even in documentation it is written it calculates fee